tracee-ebpf/tracee/tracee.go

```diff
@@ -935,8 +937,9 @@ func (t *Tracee) getProcessCtx(hostTid int) (ProcessCtx, error) {
 		if err != nil {
 			return processCtx, err
 		}
-		processCtx, err = t.ParseProcessContext(processCtxBpfMap)
-		t.processTree.processTreeMap[hostTid] = processCtx
+		containerId := t.containers.GetCgroupInfo(binary.LittleEndian.Uint64(processCtxBpfMap[8:16])).ContainerId
+		processCtx, err = processContext.ParseProcessContext(processCtxBpfMap, containerId)
```

Comment:
The fact that you now decode the container id here makes calling this parsing function irrelevant.
Instead you can pass the containers instance (t.containers) and let the function handle it internaly.